src/components/common/StyledModal.tsx

```diff
@@ -0,0 +1,172 @@
+      font-size: 1.6rem;
+      text-align: left;
+      color: var(--color-text-primary);
+      margin: 0;
+      display: flex;
+      align-items: center;
+      font-family: var(--font-default);
+      font-weight: var(--font-weight-regular);
+      border-bottom: 0.1rem solid var(--color-background-banner);
+    }
+    
+    .modali-header .modali-title {
+      font-size: inherit;
+      font-weight: inherit;
+      line-height: 1;
+    }
+    
+    .modali-header .modali-close-button {
+      background: transparent;
+      font-size: 4rem;
+      line-height: 1;
+      color: var(--color-modali-close);
+      font-weight: normal;
+      opacity: .5;
+      padding: 0;
+      margin: auto 0 auto auto;
+      
+        &:hover {
+          opacity: 1;
+        }
+    }
+    
+    .modali-body {
+      font-size: 1.4rem;
+      line-height: 1.3;
+      background: inherit;
+      color: var(--color-text-primary);
+    }
+    
+    .modali-footer {
+      margin: 1.6rem auto 0;
+      height: 5.6rem;
+      border-top: 0.1rem solid var(--color-background-banner);
+      width: 100%;
+      display: flex;
+      justify-content: space-between;
+      align-items: center;
+      padding: 0 1.6rem;
+      box-sizing: border-box;
+
+      .modali-button.modali-button-default, .modali-button.modali-button-cancel {
+        margin: 0;
+        border-radius: 0.6rem;
+        outline: 0;
+        height: 3.6rem;
+        box-sizing: border-box;
+        letter-spacing: 0.03rem;
+        display: flex;
+        align-items: center;
+        justify-content: center;
+        text-decoration: none;
+        text-transform: uppercase;
+        background: var(--color-background-CTA);
+        color: var(--color-text-CTA);
+        font-weight: var(--font-weight-bold);
+        font-size: 1.4rem;
+        
+        &:hover {
+          background-color: var(--color-background-button-hover);
+          color: var(--color-text-button-hover);
+        }
+      }
+      
+      .modali-button.modali-button-default {
+        border-radius: 0.6rem;
+        min-width: 14rem;
+        padding: 0 1.6rem;
+      }
+
+      .modali-button.modali-button-cancel {
+        background: transparent;
+        color: var(--color-text-active);
+      }
+    }
+  }
+  
+  // Heavier specification of the selector to ensure override from global.ts
+  // Ideally take out the modali default injected styles and only load from our end.
+  body.modali-open .modali.modali-size-large {
+    width: 80vw;
+    max-width: initial;
+    min-width: initial;
+    max-height: 100%;
+      @media (min-width: 500px) {
+          min-width: initial !important;
+      }
+      @media ${MEDIA.mobile},
+      @media ${MEDIA.tablet} {
+        width: 100%;
+      }
+  }
+  
+  body.modali-open .modali-wrapper-centered {
+    top: 0!important;
+  }
+  
+  body.modali-open .modali-body-style {
+    padding: 0;
+  }
+`
+
+const Modal: React.FC<Modali.ModalProps> = (props) => (
```

Comment:
An export is missing here, right?

Reply:
no its being used below in extension of `modali`

Reply:
Why do you want to export Modal when we have it inside `defaultexport.Modal`
That is if we want to follow modali compound components pattern and match modali namespace:
```js
import Modali from 'modali'
Modali.Button
Modali.Modal
```

so
```js
import Modal, { useModal, ModalHook } from 'components/common/StyledModal'
Modal.Button
Modal.Modal
```

Reply:
Is my suggestion that bad? I just find it less verbose

```js
import { Modal, useModal } from 'components/common/Modal';

const App = () => {
  const [modal, toggleModal] = useModal();

  return (
    <div>
      <button onClick={toggleModal}>Open modal</button>
      <Modal {...modal}>🔥Modal🔥</Modal>
    </div>
  );
};
```

Reply:
well to keep it as similar to the original import syntax it was kept this way. otherwise it involves changing 10+ files to use a slightly shorter syntax

Reply:
Well, the refactor is a find and replace, and the use seems to me natural. I feel that if you are abstracting the use of Modali and making a component is a fair suggestion

Not a big deal either way. So if you feel strong leave it like this. The rename I would do it (I don't like the "styled" word in our component)